src/bf_sal/linux_usr/bf_sys_dma_hugepages.c

```diff
@@ -127,8 +127,8 @@ bf_phys_addr_t bf_mem_virt2phy(const void *virtaddr) {
     close(fd);
     return BF_INVALID_PHY_ADDR;
   }
-
-  if (read(fd, &page, sizeof(uint64_t)) < 0) {
+  uint64_t size_f = read(fd, &page, sizeof(uint64_t); 
```

Comment:
This is a false positive. The return value of read call is used in the if check.

Reply:
ok Closing Pull request